pytest_timeout.py

```diff
@@ -507,3 +534,12 @@ def dump_stacks(terminal):
             thread_name = "<unknown>"
         terminal.sep("~", title="Stack of %s (%s)" % (thread_name, thread_ident))
         terminal.write("".join(traceback.format_stack(frame)))
+
+
+def pytest_runtest_makereport(item, call):
```

Comment:
this hook is apparently called many times for a single test, see `CallInfo`. Furthermore this hook is documented to "stop at first non-None result".  are you sure this hook will always correctly interact with the hook that writes to the terminal or whatever output?

Reply:
I'll add a test for `if call.when == "teardown"` so it's only called once per test.
I can also add a `hookimpl(tryfirst=True)` to try to ensure that other hooks that might return non-None run after this one. 
But really, I don't think there are guarantees